app/api/dao/task_comment.py

```diff
@@ -195,3 +196,40 @@ def delete_comment(user_id, _id, task_id, relation_id):
             return messages.TASK_COMMENT_WAS_DELETED_SUCCESSFULLY, HTTPStatus.OK
 
         return messages.TASK_COMMENT_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
+
+    @staticmethod
+    @email_verification_required
+    def report_violation(user_id: int, relation_id: int, task_id: int, _id: int):
+        """Allows a given user to report a violation for a task comment.
+
+        Args:
+            user_id: ID of the user reporting the violation
+            relation_id: ID of the relation
+            task_id: ID of the task
+            _id: ID of the comment
+
+        Returns:
+            message: A message telling the user whether the violation was successfully reported or not
+        """
+
+        is_valid = validate_data_for_task_comment(user_id, task_id, relation_id)
```

Comment:
maybe `is_invalid` is a better variable name here?

Reply:
Yes, I thought so too. However, the whole project follows the 'is_valid' naming, so 
I kept it that way. Should I change it in any case?

Reply:
Uh yes, this seems like a very appropriate change. Thank you @SanketDG !

Reply:
Done